### PR TITLE
Jokers

### PR DESCRIPTION
**Pull request by MonicaStoyanova**
50/50 and Help from audience functionalities

### CODE REVIEW

File: src/components/Jokers/Jokers.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import FiftyFiftyJoker from "./FiftyFifty/FiftyFiftyJoker";
+import AudienceHelpJoker from "./Audience/AudienceHelpJoker";
+import CallFriendJoker from "./CallFriend/CallFriendJoker";
+
+import styles from "./Jokers.module.css";
+
+const Jokers = () => {
+  return (
```
Comment (Georgi07) [inline src/components/Jokers/Jokers.js:8]: You dont need return here

File: src/components/Answers/Answers.js [hunk 2]
```diff
@@ -1,14 +1,33 @@
...
+import { useSelector } from "react-redux";
 
-const Answers = ({ shuffledAnswers, correctAnswer }) => {
+const Answers = ({ shuffledAnswers, correctAnswer, currentQuestionIndex }) => {
   const [isAnswerSelected, setIsAnswerSelected] = useState(false);
+  const [displayedAnswers, setDisplayedAnswers] = useState(shuffledAnswers);
+  const { fiftyFiftyJoker } = useSelector((state) => state.gamePlay);
 
+  useEffect(() => {
+    let answers = shuffledAnswers;
```
Comment (Georgi07) [inline src/components/Answers/Answers.js]: Unnecessary value reassignment

File: src/components/Answers/Answers.js [hunk 3]
```diff
@@ -1,14 +1,33 @@
...
+    if (
+      fiftyFiftyJoker.used &&
+      currentQuestionIndex === fiftyFiftyJoker.questionIndex
```
Comment (Georgi07) [inline src/components/Answers/Answers.js]: Declare this condition as a variable outside useEffect - "isFiftyFiftyJokerUsed"

File: src/components/BarChart/AudienceChart.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import Chart from "chart.js/auto"; // This import is needed for the chart to work properly
+import { Bar } from "react-chartjs-2";
+import "chartjs-plugin-datalabels";
+
+const AudienceChart = ({ data, options }) => {
+  return <Bar data={data} options={options} />;
+};
+
+export default AudienceChart;
```
Comment (Georgi07) [inline src/components/BarChart/AudienceChart.js]: This component is useless

File: src/components/Jokers/CallFriend/CallFriendJoker.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+} from "utils/constants";
+
+import styles from "./CallFriendJoker.module.css";
+
+const CallFriendJoker = () => {
+  const dispatch = useDispatch();
+  const { correctAnswer, callFriendJoker } = useSelector(
+    (state) => state.gamePlay
+  );
+  const [showSpeechBubble, setShowSpeechBubble] = useState(false);
```
Comment (Georgi07) [inline src/components/Jokers/CallFriend/CallFriendJoker.js]: Boolean state variables should start with "is" for example "isSpeechBubbleVisible"

File: src/components/Jokers/CallFriend/CallFriendJoker.js [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+        disabled={callFriendJoker.used}
+      ></button>
+      {showSpeechBubble && (
+        <blockquote className={styles.ovalThought}>
+          <p>
+            {FRIEND_THINKING +
+              modifiedCorrectAnswer +
+              FRIEND_WHISHING_GOOD_LUCK +
+              CLOVER_EMOJI}
+          </p>
```
Comment (Georgi07) [inline src/components/Jokers/CallFriend/CallFriendJoker.js]: Those four constants can be combined in one variable

File: src/components/Jokers/FiftyFifty/FiftyFiftyJoker.js [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+  } = useSelector((state) => state.gamePlay);
+
+  const allAnswers = [...incorrectAnswers, correctAnswer];
+
+  const handleFiftyFifty = () => {
+    if (incorrectAnswers.length < allAnswers.length / 2) {
+      return;
+    }
+
+    let reducedIncorrectAnswers = [...incorrectAnswers];
```
Comment (Georgi07) [inline src/components/Jokers/FiftyFifty/FiftyFiftyJoker.js:26]: Again unnecessary reassignment

File: src/components/Modal/Modal.js [hunk 8]
```diff
@@ -0,0 +1,21 @@
+import styles from "./Modal.module.css";
+
+const Modal = ({ isOpen, onClose, children }) => {
+  if (!isOpen) return null;
+
+  return (
+    <>
```
Comment (Georgi07) [inline src/components/Modal/Modal.js]: Unnecessary fragment tag

File: src/components/Modal/Modal.js [hunk 9]
```diff
@@ -0,0 +1,21 @@
+import styles from "./Modal.module.css";
```
Comment (Georgi07) [inline src/components/Modal/Modal.js:1]: import {overlay, modal, modalContent, closeButton} from "./Modal.module.css";

File: src/pages/Game/Game.js [hunk 10]
```diff
@@ -8,77 +8,69 @@ import GameContent from "./GameContent";
...
-            title={GAME_SCREEN_ALERTS.NO_QUESTIONS_TITLE}
-            onClose={handleAlertClose}
-          />
-        );
-
-      default:
-        return null;
+  if (status === "succeeded") {
+    if (questions.length !== 0) {
+      const currentQuestion = questions[answeredQuestionsCount];
```
Comment (Georgi07) [inline src/pages/Game/Game.js]: This variable is passed as prop to the "GameContent" component and prop name is clear enough. So we can directly assign "questions[answeredQuestionsCount]"

File: src/components/Jokers/Jokers.js (hunk 1 above)
Comment (MonicaStoyanova) [reply src/components/Jokers/Jokers.js:8]: The Jokers are not visible on the screen without the return, so we should keep it

File: src/components/Modal/Modal.js (hunk 9 above)
Comment (MonicaStoyanova) [reply src/components/Modal/Modal.js:1]: It is pure css so, it does not work like sass

File: src/components/Jokers/FiftyFifty/FiftyFiftyJoker.js (hunk 7 above)
Comment (MonicaStoyanova) [reply src/components/Jokers/FiftyFifty/FiftyFiftyJoker.js:26]: Without this assignment, if we replace the occurrence of reducedIncorrectAnswers with just incorrectAnswers, an error occurs in the for-loop:
Cannot assign to read only property '1' of object '[object Array]'

**Pull request merged**